docs/basics/basic_phases.md

```diff
@@ -0,0 +1,168 @@
+      scenario: ...
+```
+
+## rampPerSec
+
+* Similar to `constantPerSec` but ramps up or down the number of started users throughout the execution of the phase.
+
+```yaml
+Phases:
+- PHASE_NAME:
+    rampPerSec:
+      initialUsersPerSec: INITIAL_USERES_PER_SECOND
+      targetUsersPerSec: TARGET_USERES_PER_SECOND
+      duration: DURATION
+      scenario: ...
+```
+
+Example: 
+
+```yaml
+Phases:
+- rampUp:
+    rampPerSec:
+      initialUsersPerSec: 1
+      targetUsersPerSec: 100
+      duration: 5m
+      scenario: ...
+```
+
+## atOnce
+
+* All users are started when the phase starts running and once the scenario is completed the users won't retry the scenario.
+
+```yaml
+Phases:
+- PHASE_NAME:
+    atOnce:
+      users: TOTAL_USERS
+      duration: DURATION
+      scenario: ...
+```
+
+Example: 
+
+```yaml
+Phases:
+- final:
+    atOnce:
+      users: 100
+      duration: 5m
+      scenario: ...
+```
+
+## always
+
+* There is fixed number of users. Each user will start executing the scenario independently & right after the scenario is completed, start repeating the same process.  Throughput varies depending on the number of users & response time.  This is called a closed-model and is similar to the way many benchmarks with fixed number of threads work.
+
+```yaml
+Phases:
+- PHASE_NAME:
+    always:
+      users: TOTAL_USERS
+      duration: DURATION
+      scenario: ...
+```
+
+Example: 
+
+```yaml
+Phases:
+- final:
+    always:
+      users: 100
+      duration: 5m
+      scenario: ...
+```
+
+## Sequential Execution
+
+* To sequencially execute the phases, add `startAfter`
+
+
+```yaml
+Phases:
+- PHASE_NAME:
+    constantPerSec:
+      usersPerSec: USERES_PER_SECOND
+      duration: DURATION
+      startAfter: PREVIOUS_PHASE
+      scenario: ...
+```
+
+Example: 
+
+```yaml
+
+Phases:
+
+- rampUp:
+    rampPerSec:
+      initialUsersPerSec: 1
+      targetUsersPerSec: 100
+      duration: 5m
+      scenario: ...
+
+- steadyState:
+    constantPerSec:
+      usersPerSec: 100
+      duration: 5m
+      startAfter: rampUp
+      scenario: ...
+```
+
+## Sessions
+
+The state of each user's scenario is saved in the session; sometimes we speak about (re)starting sessions instead of starting new users. Hyperfoil tries to keep allocations during benchmark as low as possible and therefore it pre-allocates all memory for the scenario execution ahead. This is why all resources the benchmark uses are capped - it needs to know the sizes of pools.
+
+It is also necessary to know how many sessions we should preallocate - maximum concurrency of the system. If this threshold is exceeded Hyperfoil allocates further session as needed, but this is not the optimal mode of operation. It means that either you've underestimated the resources need or you've put a load on the system that it can't handle anymore, requests are not being completed and scenarios are not finished - which means that session objects cannot be recylced for reuse by next user.
```

Comment:
typo: recylced (I've fixed that in concepts.md)

Reply:
I removed most of them & referenced to concepts.md instead.